### ISSUE TITLE
cloud: move VPA to parent

### ISSUE THREAD

**Issue by samcday** [samcday/home-cluster#661]
The problem with running VPA inside the child cluster and not specifying any resources requests, is that if the cluster is cold-started, the poor cax11 nodes get ass-blasted by kube-scheduler. This is cos I've configured it for `MostAllocated`, and if the VPA webhook hasn't started in time, a storm of pods with no resource requests that typically use quite a lot of RAM all get scheduled  on a single node with 4GB of that stuff.

If I run VPA (or at least, the webhook) from the parent, there's a much better (guaranteed? could I configure the webhook in HA and then make sure it's fail-closed?) chance that workloads will get scheduled with the right requests.

**Comment by samcday** [comment]
This is kinda a pain in the ass. I thought I might be able to hack on the cowboysysop chart with some post renderers like I did with hccm. But the admission webhook poses a unique challenge that I haven't actually dealt with yet with a parent/child cluster setup.

I don't think it'll be particularly difficult. Mutating webhooks can be configured with a URL instead of an in-cluster service. It's just that there's enough assumptions baked into the cowboysysop chart that I'm better off forking and customizing it properly.

**Comment by samcday** [comment]
Alright, I forked the VPA chart and hacked on it enough to get this working properly.

Now the VPA components are all running in the parent cluster, the webhook config points the API server at the admission-controller running in the same namespace as the apiserver pods. Neat.

The only thing that bugs me a little is that I can't configure the webhook to be fail-closed. Or, well, I *could*, but it's too much hassle right now. This is because the Helm chart isn't reconciling that object, it's letting the admission-controller itself create it on startup, and the controller doesn't allow the `failurePolicy` of the webhook to be defined.

... Yet. I'll open an issue upstream and see if there's appetite for it. If so I'll contribute a patch since I already snooped the code and it seems like it ought to be straightforward to do so.

I'll close this issue out as done now, though. Since the previous setup was also fail-open. So I haven't made anything worse, but actually it should already be "good enough" now - the admission controller is running 2 replicas in the parent with a PDB, so it should be quite rare indeed for a pod to schedule in the cloud-cluster at a moment where no VPA admission-controllers in the parent are able to handle the request.
